coalib/processes/Processing.py

```diff
@@ -513,17 +514,30 @@ def yield_ignore_ranges(file_dict):
                                    len(file[line_number-1])))
 
                 else:
-                    for ignore_stmt in ['ignore ', 'noqa ', 'noqa']:
+                    for ignore_stmt in ['ignore\n', 'noqa\n',
+                                        'noqa ', 'ignore ']:
                         if ignore_stmt in line:
-                            end_line = min(line_number + 1, len(file))
-                            yield (get_ignore_scope(line, ignore_stmt),
-                                   SourceRange.from_values(
-                                       filename,
-                                       line_number, 1,
-                                       end_line, len(file[end_line-1])))
+                            manip_line = re.sub('[^A-Za-z0-9]+', '', line)
+                            if (manip_line.strip()).startswith(
+                              ignore_stmt.strip()):
+                                end_line = min(line_number + 1, len(file))
+                            else:
+                                end_line = line_number
+                            if ignore_stmt.endswith('\n'):
+                                yield ([],
+                                       SourceRange.from_values(
+                                           filename,
+                                           line_number, 1,
+                                           end_line, len(file[end_line-1])-1))
+                            else:
+                                yield (get_ignore_scope(line, ignore_stmt),
+                                       SourceRange.from_values(
+                                           filename,
+                                           line_number, 1,
+                                           end_line, len(file[end_line-1])-1))
                             break
 
-        if stop_ignoring is False and start is not None:
+        if stop_ignoring is True and start is not None:
```

Comment:
Is this correct? Just want to confirm that this is intentional, this looks so, "groundbreaking" :D

Reply:
🤪 That was indeed ground breaking. Made that during testing something